core/blockchain.go

```diff
@@ -1053,6 +1057,33 @@ func (bc *BlockChain) Accept(block *types.Block) error {
 	bc.addAcceptorQueue(block)
 	acceptedBlockGasUsedCounter.Inc(int64(block.GasUsed()))
 	acceptedTxsCounter.Inc(int64(len(block.Transactions())))
+	return bc.handlePrecompilePostAccept(block)
+}
+
+// handlePrecompilePostAccept handles executing the post-accept functions of any active precompiles
+// on the logs contained in the block.
+func (bc *BlockChain) handlePrecompilePostAccept(block *types.Block) error {
```

Comment:
Out of scope for this PR, but we need to ensure that this provides the atomicity required by the precompiles and works correctly with how we handle a potential shutdown interrupting this operation, such that we resume and call accept on the same block if the VM did not finish marking it as accepted.